pkg/logs/input/container/filter.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build !windows
+
+package container
+
+import (
+	"encoding/json"
+	"strings"
+
+	log "github.com/cihub/seelog"
+	"github.com/docker/docker/api/types"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+)
+
+// Container represents a container to tail logs from.
+type Container struct {
+	Identifier string
+	Image      string
+	Source     *config.LogSource
+}
+
+// NewContainer returns a new Container
+func NewContainer(container types.Container, source *config.LogSource) *Container {
+	return &Container{
+		Identifier: container.ID,
+		Image:      container.Image,
+		Source:     source,
+	}
+}
+
+// Filter returns all the containers for which a source exists.
+func Filter(containers []types.Container, sources []*config.LogSource) []*Container {
+	containersToTail := []*Container{}
+	for _, container := range containers {
+		if source := searchSource(container, sources); source != nil {
+			containersToTail = append(containersToTail, NewContainer(container, source))
+		}
+	}
+	return containersToTail
+}
+
+// searchSource returns a valid source for container,
+// if no source is found, return nil
+func searchSource(container types.Container, sources []*config.LogSource) *config.LogSource {
+	if source := sourceFromContainer(container); source != nil {
+		return source
+	}
+	var candidate *config.LogSource
+	for _, source := range sources {
+		if source.Config.Image != "" && !isImageMatch(source.Config.Image, container.Image) {
+			continue
+		}
+		if source.Config.Label != "" && !isLabelMatch(source.Config.Label, container.Labels) {
+			continue
+		}
+		if candidate == nil {
+			candidate = source
+		}
+		if computeScore(container, candidate) < computeScore(container, source) {
+			candidate = source
+		}
+	}
+	return candidate
+}
+
+// computeScore returns the matching score between the container and the source,
+// the higher, the better
+func computeScore(container types.Container, source *config.LogSource) int {
```

Comment:
Same as above: What prevents you from defining this func directly on a Container ?